jobs/wf1-2-createPatients.js

```diff
@@ -0,0 +1,102 @@
+//Define gender options and prepare newPatientUuid and identifiers
+fn(state => {
+  const genderOptions = {
+    male: 'M',
+    female: 'F',
+    unknown: 'U',
+    'Transgender female': 'O',
+    'Transgender male': 'O',
+    'Prefer not to answer': 'O',
+    'Gender variant - Non conforming': 'O',
+  };
+
+  const identifiers = [];
+  const newPatientUuid = [];
+
+  return { ...state, genderOptions, newPatientUuid, identifiers };
+});
+
+// Creating identifiers for each patient
+each('trackedEntityInstances[*]', state => {
+  return post(
+    'idgen/identifiersource/8549f706-7e85-4c1d-9424-217d50a2988b/identifier',
+    {}
+  )(state).then(state => {
+    state.identifiers.push(state.data.body.identifier);
+    return state;
+  });
+});
+
+// Map trackedEntityInstances to openMRS data model
+fn(state => {
+  const { trackedEntityInstances, identifiers, genderOptions } = state;
+
+  const pluckAttributeValue = (arr, keyVal) => {
+    const result = arr.filter(a => a.code === keyVal);
+    return result.length > 0 ? result[0].value : undefined;
+  };
```

Comment:
Instead of filtering, `Array.prototype.find` does the same thing, with the added bonus of returning as soon as it finds a match.

```js
const getValueForCode = (attributes, code) => {
  return attributes.find((attribute) => attribute.code === code)?.value;
};

const examples = [{ code: "foo", value: "bar" }];

console.log(getValueForCode(examples, "foo"));  // bar
console.log(getValueForCode(examples, "baz"));  // undefined
```

Reply:
I have notice the `?.value` does not work, Is this a nodejs feature ?